### PR TITLE
8 add comment

### PR DESCRIPTION
**Pull request by roseannac21**

### CODE REVIEW

**Comment by roseannac21** [comment]
just been in Connor's afternoon zoom chat with a few others and seen that it's possible to do post request without a useEffect - wondering now if the way I have done this is not optimal/most efficient? Maybe though i could come back to it if i finish all necessary tasks before the end of Friday?

File: src/components/ReviewExtraInfo.jsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const ReviewExtraInfo = () => {
             setReview(data)
             setLikeCount(data.votes)
         })
-    }, []);
+    }, [review_id]);
 
     if (isLoading) {
         return <p>Loading review...</p>
```
Comment (elijaharley) [inline src/components/ReviewExtraInfo.jsx:25]: This loading state needs to be declared/passed as props

File: src/components/PostComment.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import { useEffect } from "react";
+import { useState } from "react";
+import axios from "axios";
+
+const PostComment = ({review_id, setComments, setPostSuccessful}) => {
+
+const [commentData, setCommentData] = useState({
+    author: "jessjelly",
+    body: "" 
+});
```
Comment (elijaharley) [inline src/components/PostComment.jsx:10]: Having a user in state is useful here until you have a user context - don't forget to update it when you add user context.

File: src/components/PostComment.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            setCommentData({...commentData, body:event.target.value})
+        }} />
+    <button>Add comment</button>
+    </form>
+    </>
+)
+
+}
+
+export default PostComment
```
Comment (elijaharley) [inline src/components/PostComment.jsx:66]: You may want your placeholder text to disappear when clicked to avoid confusion.

**Review by elijaharley — APPROVED** [review]
I've left a couple of comments inline, it's a good question about useEffect - I think that the method that you've used works well so there is no need to change it, but if you want to after the essential endpoints then you can do. 🤩

**Pull request merged**